### PR TITLE
Fix compare modal bug

### PR DESCRIPTION
**Pull request by chasulin**

### CODE REVIEW

**Comment by chasulin** [comment]
[Bug: Compare table shows only one county when selecting USA (county page, modal)](https://trello.com/c/QeLzbHoa/1152-bug-compare-table-shows-only-one-county-when-selecting-usa-county-page-modal)

File: src/components/Compare/LocationTable.tsx [hunk 1]
```diff
@@ -244,11 +244,11 @@ const LocationTable: React.FunctionComponent<{
 
   // In the modal, if the rank of the pinned-location-row is #1, we remove
   // the location's inline row, so as to not have the location listed twice consecutively:
-  const removePinnedIfRankedFirst = (location: SummaryForCompare) =>
+  const removeInlineIfRankedFirst = (location: SummaryForCompare) =>
```
Comment (chasulin) [inline src/components/Compare/LocationTable.tsx:247]: name swap feels more intuitive

File: src/components/Compare/LocationTable.tsx [hunk 2]
```diff
@@ -257,11 +257,11 @@ const LocationTable: React.FunctionComponent<{
     } else {
       if (region) {
         if (geoScope === GeoScopeFilter.COUNTRY)
-          return sortedLocations.slice(0, 100);
+          return modalLocations.slice(0, 100);
         else return modalLocations;
       } else {
         if (homepageScope === HomepageLocationScope.STATE)
-          return modalLocations;
+          return sortedLocations;
```
Comment (chasulin) [inline src/components/Compare/LocationTable.tsx:264]: this (prev. set as `modalLocations`) was resolving to `sortedLocations`, so i just replaced `modalLocations` with `sortedLocations`

File: src/components/Compare/LocationTable.tsx (hunk 1 above)
Comment (pnavarrc) [reply src/components/Compare/LocationTable.tsx:247]: isNotPinnedLocation

**Review by pnavarrc — APPROVED** [review]

**Pull request merged**